mypyc/test-data/run-strings.test

```diff
@@ -178,3 +178,22 @@ def test_str_to_bool() -> None:
     for x in 'a', 'foo', 'bar', 'some string':
         assert is_true(x)
         assert not is_false(x)
+
+[case testStringFormatting]
+
+def test_format_method() -> None:
+    name = "Eric"
+    age = 14
+
+    assert "Hello, {}!".format(name) == "Hello, Eric!"
+    assert "{}".format(name) == "Eric"
+    assert "{}! Hi!".format(name) == "Eric! Hi!"
+    assert "{}, Hi, {}".format(name, name) == "Eric, Hi, Eric"
+    assert "Hi! {}".format(name) == "Hi! Eric"
+    assert "Hi, I'm {}. I'm {}.".format(name, age) == "Hi, I'm Eric. I'm 14."
+
+    assert "{{}}{}".format(name) == "{}Eric"
+    assert "Hi! {{{}}}".format(name) == "Hi! {Eric}"
+    assert "Hi! {{ {}".format(name) == "Hi! { Eric"
+    assert "Hi! {{ {} }}}}".format(name) == "Hi! { Eric }}"
```

Comment:
Please add more tests. We want fairly complete test coverage to make sure nothing has regressed. It's important both valid and error cases produce the same output when optimized. For example, any exceptions should be identical.

Examples (but there are probably more):
* Simplest possible case: `''.format()`
* Plain text only: `'xyz'.format()`
* Extra positional arguments: `'a{}b'.format(1, 2, 3)`
* Too few arguments
* Various format specifiers that are supported within `{...}` (these should still work, even if they aren't optimized)
* Keyword arguments to `format`
* `*args` to `format`
* Invalid format string

Reply:
Extra positional arguments case this compile error: `error: Not all arguments converted during string formatting`

Reply:
These test cases could still be useful. Can you double check if they aren't covered:
* `'{}{}'.format(1, 2)` (empty literals)
* `'x{}y{}'.format(*args)` (`*args`)